core/src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -815,8 +883,7 @@ object MergeIntoCommand {
       val fromRow = outputRowEncoder.createDeserializer()
       rowIterator
         .map(toRow)
-        .map(processRow)
-        .filter(!shouldDeleteRow(_))
+        .flatMap(processRow)
         .map { notDeletedInternalRow =>
```

Comment:
where is this filter gone?

Reply:
Since we are now generating a sequence of output rows for each internal input row, we no longer need `ROW_DELETED_COL` column since we can simply omit dropped rows from the initial sequence.

Before, we had to have one output row for each clause. So for example in the case of the delete clause, we had to output a row that later needed to be drop. Now, we can simply output `Seq()`.